src/sql/workbench/contrib/searchViewPane/browser/searchWidget/searchWidgetService.ts

```diff
@@ -0,0 +1,297 @@
+		@IInstantiationService private readonly _instantiationService: IInstantiationService,
+		@IViewDescriptorService protected viewDescriptorService: IViewDescriptorService,
+	) {
+		super();
+		this.searchWidgets = new Map<string, any>();
+	}
+
+	getSearchWidget(VIEWLET_ID: string): any | undefined {
+		return this.searchWidgets.get(VIEWLET_ID);
+	}
+
+	registerSearchWidget(container: HTMLElement, pane: ViewPaneContainer, params: ISearchViewPaneOptions): void {
+		if (!this.searchWidgets.get(params.VIEWLET_ID)) {
+			let searchWidget = this._register(this._instantiationService.createInstance(SimpleSearchWidget, container, <IViewExplorerSearchOptions>{
+				value: '',
+				replaceValue: undefined,
+				isRegex: false,
+				isCaseSensitive: false,
+				isWholeWords: false,
+				searchHistory: [],
+				replaceHistory: [],
+				preserveCase: false,
+				showSearchResultsPane: params.showSearchResultsPane,
+			}, pane, params.VIEWLET_ID));
+
+			this.searchWidgets.set(params.VIEWLET_ID, searchWidget);
+			if (params.showSearchResultsPane) {
+				this.registerSearchResultsView(params);
+
+				//register search results view specific stuff.
+				if (searchWidget.searchView) {
+					this._register(searchWidget.onDidHeightChange(() => searchWidget.searchView?.reLayout()));
+
+					this._register(searchWidget.onPreserveCaseChange(async (state) => {
+						if (searchWidget.searchView?.searchViewModel) {
+							searchWidget.searchView.searchViewModel.preserveCase = state;
+							await searchWidget.searchView.refreshTree();
+						}
+					}));
+
+					this._register(searchWidget.searchInput.onInput(() => searchWidget.searchView?.updateActions()));
+				}
+			}
+
+			if (params.onSearchSubmit) {
+				searchWidget.onSearchSubmit(options => params.onSearchSubmit(options));
+
+				searchWidget.searchInput.onDidOptionChange((options) => params.onSearchSubmit(options));
+			}
+
+			if (params.onSearchCancel) {
+				searchWidget.onSearchCancel((focus) => params.onSearchCancel(focus));
+			}
+		}
+	}
+
+	registerSearchResultsView(params: ISearchViewPaneOptions) {
+		let viewDescriptors = [];
+		switch (params.VIEWLET_ID) {
+			case 'workbench.view.notebooks': viewDescriptors.push({
+				id: NotebookSearchResultsView.ID,
+				name: localize('notebookExplorer.searchResults', "Search Results"),
+				ctorDescriptor: new SyncDescriptor(NotebookSearchResultsView),
+				weight: 100,
+				canToggleVisibility: true,
+				hideByDefault: false,
+				order: 0,
+				collapsed: true
+
+			});
+				break;
+			default: break;
+		}
+
+		const container = this.viewDescriptorService.getViewContainerById(params.VIEWLET_ID);
+		Registry.as<IViewsRegistry>(ViewContainerExtensions.ViewsRegistry).registerViews(viewDescriptors, container);
+	}
+
+	getSearchResultsView(parentConatiner: ViewPaneContainer): ISearchResultsView {
+		switch (parentConatiner.getId()) {
+			case 'workbench.view.notebooks': return <NotebookSearchResultsView>parentConatiner.getView(NotebookSearchResultsView.ID);
```

Comment:
Why does this have to know about the containers that are using it? Shouldn't that be something you capture from registerSearchResultsView?